torch/_higher_order_ops/flex_attention.py

```diff
@@ -478,6 +479,20 @@ def flex_attention_fake_tensor_mode(
         return out, logsumexp
 
 
+@flex_attention.py_impl(_CachingTorchDispatchMode)
+def _flex_caching(
+    mode: _CachingTorchDispatchMode, *args: Any, **kwargs: Any
+) -> tuple[torch.Tensor, torch.Tensor]:
+    return mode.__torch_dispatch__(flex_attention, (), args, kwargs)
```

Comment:
@drisspg can you make a new API called torch._higher_order_ops.utils.redirect_to_mode(HOP, TorchDispatchMode) that expands to this registration? Then you can use it here twice. 

Can also replace https://github.com/pytorch/pytorch/blob/1c6e88eb0330a6b3fecf6033b0ffc55a11f181be/torch/_higher_order_ops/utils.py#L785-L788 with that API.

This is something pretty common. We could also go as far as to recommend this in our HOP x TorchDispatchMode error messages.

Reply:
Maybe even just make it HOP.redirect_mode(TorchDispatchMode)

Reply:
addded